pkg/pipeline/instance.go

```diff
@@ -58,10 +58,20 @@ type Instance struct {
 	ConnectorIDs []string
 	ProcessorIDs []string
 
+	status     Status
+	statusLock sync.RWMutex
```

Comment:
@hariso do you think this lock would be relevant only to status or we could have the opportunity to have a lock for anything that would mutate the pipeline? Something tells me that this is basically it, but maybe with the need of a broader naming? What do you think?

For context, this line made me think of that https://github.com/ConduitIO/conduit/pull/1794/files#diff-576b2656f57e455d4f1e5405fd78db341ddef5d190447eb2e15b5088bacc0980R96.

Reply:
I imagine we can do it! We would need to think about what exactly needs to be locked since there might be cases where we don't want to lock the whole pipeline for a single field.